docs/views_and_forms.rst

```diff
@@ -120,6 +125,41 @@ Views
 
 .. currentmodule:: haystack.views
 
+.. note::
+
+    As of version 2.4, the old ``haystack.views.SearchView`` will be deprecated in favor of the new generic, django-style search views at ``haystack.generic_views.SearchView``.
+
+New Django-Style Class Based Views
+----------------------------------
+
+Version 2.4 introduced familiar django-style class base views which extend from 
+django's ``FormView``.  The new views are located at 
+``haystack.generic_views.SearchView``.  The generic search view follows the 
+common view patterns laid out by django and can be extended like any other 
+view::
+    
+    # views.py
+    
+    from haystack.generic_views import SearchView
+    
+    class MySearchView(SearchView):
+        """My custom search view."""
+        
+        def get_context_data(self, *args, **kwargs):
+            context = super(MySearchView, self).get_context_data(*args, **kwargs)
+            # do something
```

Comment:
Do you think it'd be worth including an example here which would show how to e.g. filter the SearchQuerySet used? That's a really common task